jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -525,3 +526,36 @@ def _live_update(self, event={}):
         for mark in self.marks.values():
             mark.update_xy(sp.spectral_axis.value, sp.flux.value)
             mark.visible = True
+
+    def translate_units(self, spectral_cube, collapsed_spec):
+        if collapsed_spec._unit == u.MJy / u.sr:
+            # MJy to Jy
+            collapsed_spec._data *= 10e6
+
+            # spectral_cube.meta['_pixel_scale_factor'] / 1, where 1 = default scale factor
+            collapsed_spec._data *= spectral_cube.meta['_pixel_scale_factor']
```

Comment:
I think the scale factors will need to be stored in `collapsed_spec` instead of `spectral_cube`, since a single spectral cube can be used to extract multiple spectra with different subsets/apertures.

Reply:
You are right, that makes a lot more sense instead of the flux data!